File: schemas/safety.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime
from app.domains.enums import IncidentStatus
from app.domains.enums import IncidentSeverity

class SafetyIncidentCreate(BaseModel):
    severity: IncidentSeverity
    description: str | None = None
    location: str
    department: str
    shift: str

class SafetyIncidentUpdate(BaseModel):
    severity: IncidentSeverity | None = None
    description: str | None = None
    location: str | None = None
    department: str | None = None
    shift: str | None = None
    corrective_action: str | None = None

class SafetyIncidentOut(BaseModel):
    id: UUID
    severity: IncidentSeverity
    description: str | None
    location: str
    department: str
    shift: str
    status: IncidentStatus
    corrective_action: str | None
    reported_by: UUID
    created_at: datetime

    class Config:
        from_attributes=True

File: frontend/src/types/safety.ts
export type IncidentStatus = "open" | "investigating" | "corrective_action" | "closed"; 
export type IncidentSeverity = "near_miss" | "soreness" | "first_aid" | "recordable" | "psif";

export interface SafetyIncidentCreate {
    severity: IncidentSeverity;
    description?: string;
    location: string;
    department: string;
    shift: string;
}

export interface SafetyIncidentUpdate {
    severity?: IncidentSeverity;
    description?: string;
    location?: string;
    department?: string;
    shift?: string;
    corrective_action?: string;
}

export interface SafetyIncidentOut {
    id: string;
    severity: IncidentSeverity;
    description?: string;
    location: string;
    department: string;
    shift: string;
    status: IncidentStatus;
    corrective_action?: string;
    reported_by: string;
    created_at: string;
}

export interface SafetyIncidentAuditLog {
    id: string;
    incident_id: string;
    changed_by: string;
    old_status: IncidentStatus;
    new_status: IncidentStatus;
    changed_at: string;
}

export interface SafetyIncidentStatusUpdate {
    status: IncidentStatus;
}

